### PR TITLE
CSP file ensuring the app scripts & styles are handled

### PR DESCRIPTION
**Pull request by tahir-khalid**
## Description

Fixed shared partial that loads the GTM related scripts with a `javascript_tag` helper to ensure `nonce` is handled and resolve the CSP violations currently in Staging.
Resolved GA container script and source related issues.

### CODE REVIEW

File: app/views/shared/_ga_head.html.erb [hunk 1]
```diff
@@ -1,14 +1,14 @@
-<script>
-    window.dataLayer = window.dataLayer || [];
-    dataLayer.push({
-      'userId': <%= Digest::SHA256.hexdigest(current_user&.id.to_s) if current_user %>
-    });
- </script>
+<%= javascript_tag nonce: true do %>
+  window.dataLayer = window.dataLayer || [];
+  dataLayer.push({
+    'userId': '<%= Digest::SHA256.hexdigest(current_user&.id.to_s) if current_user %>'
```
Comment (meeravnagesh) [inline app/views/shared/_ga_head.html.erb:4]: dint the GA team mention they can unhex the code ?

File: config/initializers/content_security_policy.rb [hunk 2]
```diff
@@ -2,9 +2,13 @@
 
 # Trusted sources
 trusted_script_sources = [:self, "https://www.googletagmanager.com"]
-trusted_style_sources  = [:self, "https://fonts.googleapis.com"]
+trusted_style_sources = [:self, "https://fonts.googleapis.com"]
 trusted_font_sources = [:self, "https://fonts.gstatic.com", "*.gov.uk"]
 trusted_img_sources = %i[self data]
+trusted_connect_sources = [:self,
+                           "https://www.google-analytics.com",
```
Comment (meeravnagesh) [inline config/initializers/content_security_policy.rb:9]: "https://www.google-analytics.com",
                           "https://*.google-analytics.com", are these two not inclusive

File: app/views/shared/_ga_head.html.erb (hunk 1 above)
Comment (tahir-khalid) [reply app/views/shared/_ga_head.html.erb:4]: Correct not sure how that got back in there reverted to plain UUID value.

File: config/initializers/content_security_policy.rb (hunk 2 above)
Comment (tahir-khalid) [reply config/initializers/content_security_policy.rb:9]: Correct I have removed the explicit value.

**Review by meeravnagesh — APPROVED** [review]

**Pull request merged**